app/controllers/posts_controller.rb

```diff
@@ -1,11 +1,44 @@
 class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
+    @posts = @user.posts
   end
 
   def show
     @user = User.find(params[:user_id])
     @post = @user.posts[params[:id].to_i - 1]
-    @post_idx = @user.posts.find_index(@post) + 1
+    @post_idx = params[:id]
+  end
+
+  def new
+    @user = User.find(params[:user_id])
+    @post = @user.posts.new
+
+    render :new, locals: { post: @post }
+  end
+
+  def create
+    @user = User.find(params[:user_id])
+    @post = @user.posts.new(post_params)
```

Comment:
- [ ] this is what you created the `current_user` method for in the application_controller, you can write it as follows: `@post = @current_user.posts.new(post_params)`